xilem_core/examples/filesystem.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2024 the Xilem Authors
```

Comment:
It's really nice that you added examples that don't require understanding Masonry, but I think what you actually want is unit tests.

Reply:
Yeah, definitely. These models were mostly to test that things compiled as I was building up this core.

Do you want to block the PR on these being added?

Reply:
Honestly, unless this PR is blocking further work (and even then), I'd say yes, until at least one test is added, for good practice.

We don't have to get 100% coverage, but writing tests in the same PR as the code is an habit you have to build, and now is as good a time as any to start. And It's a good precedent for this codebase.

Reply:
Great. For some context on this untested state, the final version of this pr was submitted as my final action before going on holiday.

My first task for next week can certainly be adding tests here.